### PR TITLE
Merchant e-Solutions: pass optional 3Dsecure params

### PR DESCRIPTION
**Pull request by curiousepic**
@duff to review

### CODE REVIEW

File: lib/active_merchant/billing/gateways/merchant_e_solutions.rb [hunk 1]
```diff
@@ -122,6 +125,15 @@ def add_creditcard(post, creditcard, options)
         post[:card_exp_date]  = expdate(creditcard)
       end
 
+      # not a 3dsecure integration - simply passes params to gateway if present,
+      # verified externally
+      def add_3dsecure_params(post, options)
+        post[:xid] = options[:xid] unless options[:xid].blank?
```
Comment (duff) [inline lib/active_merchant/billing/gateways/merchant_e_solutions.rb]: We're trying to lessen our dependence on ActiveSupport so we're trying to lessen the use of `blank?`.  Example of the approach we're using:

https://github.com/activemerchant/active_merchant/blob/master/lib/active_merchant/billing/gateways/authorize_net.rb#L463
https://github.com/activemerchant/active_merchant/blob/master/lib/active_merchant/billing/gateways/authorize_net.rb#L6

File: lib/active_merchant/billing/gateways/merchant_e_solutions.rb [hunk 2]
```diff
@@ -122,6 +125,15 @@ def add_creditcard(post, creditcard, options)
         post[:card_exp_date]  = expdate(creditcard)
       end
 
+      # not a 3dsecure integration - simply passes params to gateway if present,
```
Comment (duff) [inline lib/active_merchant/billing/gateways/merchant_e_solutions.rb]: I'd recommend removing the comments.

**Comment by duff** [comment]
@curiousepic Looks good.  After the few little tweaks I mentioned, we should be good to go.

If you force push your branch with the adjustments, the PR should reflect it.

**Comment by curiousepic** [comment]
@duff PR updated with those changes.

**Pull request closed without merging**